contracts/lazyMint/LazyMintERC1155.sol

```diff
@@ -0,0 +1,606 @@
+    /// @dev Token ID => public mint conditions for tokens with that ID.
+    mapping(uint256 => PublicMintConditions) public mintConditions;
+    /// @dev Token ID => the address of the recipient of primary sales.
+    mapping(uint256 => address) public saleRecipient;
+
+    /// @dev Checks whether caller has DEFAULT_ADMIN_ROLE on the protocol control center.
+    modifier onlyProtocolAdmin() {
+        require(
+            controlCenter.hasRole(controlCenter.DEFAULT_ADMIN_ROLE(), _msgSender()), 
+            "LazyMintERC1155: not protocol admin."
+        );
+        _;
+    }
+
+    /// @dev Checks whether caller has DEFAULT_ADMIN_ROLE.
+    modifier onlyModuleAdmin() {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "LazyMintERC1155: not module admin.");
+        _;
+    }
+
+    /// @dev Checks whether caller has MINTER_ROLE.
+    modifier onlyMinter() {
+        require(hasRole(MINTER_ROLE, _msgSender()), "LazyMintERC1155: not minter.");
+        _;
+    }
+
+    constructor(
+        string memory _contractURI,
+        address payable _controlCenter,
+        address _trustedForwarder,
+        address _nativeTokenWrapper,
+        address _saleRecipient
+    ) 
+        ERC1155("")
+        ERC2771Context(_trustedForwarder)
+    {
+        
+        controlCenter = ProtocolControl(_controlCenter);
+        nativeTokenWrapper = _nativeTokenWrapper;
+        defaultSaleRecipient = _saleRecipient;
+        contractURI = _contractURI;
+
+        address deployer = _msgSender();
+        _setupRole(DEFAULT_ADMIN_ROLE, deployer);
+        _setupRole(MINTER_ROLE, deployer);
+        _setupRole(TRANSFER_ROLE, deployer);
+    }
+
+    ///     =====   Public functions  =====
+
+    /// @dev Returns the URI for a given tokenId.
+    function uri(uint256 _tokenId) public view override returns (string memory _tokenURI) {
+
+        for(uint256 i = 0; i < baseURIIndices.length; i += 1) {
+            if(_tokenId < baseURIIndices[i]) {
+                return string(abi.encodePacked(baseURI[baseURIIndices[i]], _tokenId.toString()));
+            }
+        }
+        
+        return "";
+    }
+
+    ///     =====   External functions  =====
+
+    /**
+     *  @dev Lets an account with `MINTER_ROLE` mint tokens of ID from `nextTokenIdToMint` 
+     *       to `nextTokenIdToMint + _amount - 1`. The URIs for these tokenIds is baseURI + `${tokenId}`.
+     */
+    function lazyMint(uint256 _amount, string calldata _baseURIForTokens) external onlyMinter {
+        uint256 startId = nextTokenIdToMint;
+        uint256 baseURIIndex = startId + _amount;
+
+        nextTokenIdToMint = baseURIIndex;
+        baseURI[baseURIIndex] = _baseURIForTokens;
+        baseURIIndices.push(baseURIIndex);
+        
+        emit LazyMintedTokens(startId , startId + _amount - 1, _baseURIForTokens);
+    }
+    
+    /// @dev Lets an account claim a given quantity of tokens, of a single tokenId.
+    function claim(
```

Comment:
given we're using multicall, make sure you write tests for multicall + claim and making sure we don't bypass some mint conditions just because it's part of multicall